core/src/main/java/org/apache/calcite/adapter/jdbc/JdbcTable.java

```diff
@@ -200,6 +200,24 @@ public RelNode toRel(RelOptTable.ToRelContext context,
         sourceExpressionList, flattened);
   }
 
+  @Override public boolean equals(Object obj) {
```

Comment:
even though this is trying to fix CALCITE-3514, because this is a general purpose `equals` implementation, it should checks equality of all fields of this class. Same for hashcode.

Reply:
I disagree. Two tables are identical if they have the same name, schema etc. No need to check columns.

The "equals" contract does not say you have to check every field.

Reply:
O I see. So if schema name is the same then it guarantees that two tables will be the same (e.g. no need to check the schema), in that case it would make sense.

Reply:
I add a variable in `JdbcSchema` to trigger the issue.